privacyidea/lib/log.py

```diff
@@ -141,12 +141,18 @@ def __call__(self, func):
         def log_wrapper(*args, **kwds):
             """
             Wrap the function in log entries. The entry of the function and
-            the exit of the function is logged.
+            the exit of the function is logged using the DEBUG log level.
+            If the logger does not log DEBUG messages, this just returns
+            the result of ``func(*args, **kwds)`` to improve performance.
 
             :param args: The positional arguments starting with index
             :param kwds: The keyword arguemnts
             :return: The wrapped function
             """
+            # Exit early if self.logger disregards DEBUG messages.
+            if not self.logger.isEnabledFor(logging.DEBUG):
```

Comment:
I see. I forgot that the whole log_with will only have effect if the loglevel is ``<= logging.DEBUG.``

But please note, that we also recommend setting the loglevel to ``9`` which is no official log level but is used to log passwords to the log file. We use this here:

                    level = self.logger.getEffectiveLevel()
                    # Check if we should not do the password logging.
                    # I.e. we only do password logging if log_level < 10.
                    if level != 0 and level >= 10:
                        # Hide specific arguments or keyword arguments
                        log_args = deepcopy(args)
                        log_kwds = deepcopy(kwds)

So can you confirm that the log level 9 also matches ``self.logger.isEnabledFor(logging.DEBUG)``?